### DISCUSSION TITLE
Channels to an existing WSGI application

### DISCUSSION THREAD

**Discussion by jet2018** [django/channels#1829] [Ideas]
So, hello great team, am a beginner to using Django channels so if my request was already handled, please inform me. 
Otherwise, to add channels to an existing application, one needs to change the application server from WSGI to ASGI, this at some point breaks the application, yet someone wants to send a little few push notifications. So can't there be a minimal way of serving the channels with existing application without changing the WSGI. Maybe to serve minimal services, so that the advanced ASGI can be used with the complex ones.

Please sorry for being somehow verbose, but I can try to make more clarifications if needed.
Thanks

**Comment by carltongibson** [comment]
Hi @jet2018 — good question. 

So, I'd just keep your WSGI app as it is. Then have your reverse proxy or load balancer (or ...) route just some paths (or `wss://` requests) to an ASGI app that you run on the side. This can just handle the few async bits as you're adding them. 

Does that make sense?

**Comment by jet2018** [comment]
Thanks, but what i feel like seeing is when I can emit an event amidst my
requests without going out of the views to first configure ASGI for
something really small like that.

On Thu, Mar 17, 2022 at 11:55 AM Carlton Gibson ***@***.***>
wrote:

> Hi @jet2018 <https://github.com/jet2018> — good question.
>
> So, I'd just keep your WSGI app as it is. Then have your reverse proxy or
> load balancer (or ...) route just some paths (or wss:// requests) to an
> ASGI app that you run on the side. This can just handle the few async bits
> as you're adding them.
>
> Does that make sense?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/django/channels/discussions/1829#discussioncomment-2379029>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKJ2IPCE7BYJ3LBL2MHQS33VALXOVANCNFSM5Q6E6ARA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
jet

**Reply by carltongibson** [reply]
Well you could use a [StreamingHttpResponse](https://docs.djangoproject.com/en/4.0/ref/request-response/#django.http.StreamingHttpResponse) — but see the admonition there — WSGI just isn't built for that kind of thing. 

The other option is polling... make a request every X seconds to see if there's an update. That's perfectly fine at small (and even medium) scales. 

> ...something really small like that

The whole motivation for async is to be able to address these use-cases...
